internal/guest/bridge/bridge.go

```diff
@@ -366,6 +366,11 @@ func (b *Bridge) ListenAndServe(bridgeIn io.ReadCloser, bridgeOut io.WriteCloser
 	go func() {
 		var resperr error
 		for resp := range b.responseChan {
+			var err error
+			_, span := oc.StartSpan(context.Background(), "request write response", oc.WithClientSpanKind)
+			defer span.End()
+			defer func() { oc.SetSpanStatus(span, err) }()
```

Comment:
As @helsaawy mentioned above, won't these run only when the response go routine exits? Which I think is the bridge response loop and it won't exit until gcs exits or we hit some of the errors below? Won't these spans just wait for the go routine to end and not even sure if they'll have enough time to send anything over the bridge connection? You'd probably need to wrap the loop body inside a func to make sure the defered calls succeed? e.g. https://go.dev/play/p/JhzoFYRsXzS

Reply:
on a second look, the error will already be recorded. Like the trace log here, and in the functions above, I think logging a span with message type atleast for the success cases would be important as the complete message itself is too long and consumes a large size. Let me know if the newly added change looks good.

Reply:
I dont think we need the span here at all, we already have a span thats started on L272 and lives for the entire request/response timeline
plus, the defers won't be run until the end of the goroutine, which is only when the bridge is closed, so we'll only pile up a lot of spans in limbo and then close them all at once, with wrong end times and durations